weave/logs.py

```diff
@@ -26,7 +26,7 @@ class LogFormat(enum.Enum):
 @dataclasses.dataclass
 class LogSettings:
     format: LogFormat
-    level: str
+    level: typing.Optional[int]
```

Comment:
To eliminate the need for converting back/forth between `str` and `int` log levels, unify everything to `int`.

`None` here is allowed as a "just follow the logger's level" option.